### ISSUE TITLE
Column Definition Changed to NUMERIC from DECIMAL

### ISSUE THREAD

**Issue by mab05k** [doctrine/dbal#3403]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.9.0

#### Summary

After upgrading to v2.9, running `doctrine:schema:validate` command fails. Recommends changing my column definition from DECIMAL to NUMERIC.

#### Previous behaviour

Previous column definition
```sql
`column_name` DECIMAL(6,3) NOT NULL DEFAULT '1.000'
```

#### Current behavior

New Column Definition
```sql
`column_name` NUMERIC(6, 3) NOT NULL DEFAULT '1.000'
```

#### How to reproduce
This configuration was being used in a Symfony Application
Using MySQL 5.6
Run `bin/console doctrine:schema:create` with v2.8
Entity Mapping Configuration
```yaml
        column:
            type: decimal
            nullable: false
            precision: 6
            scale: 3
            column: column_name
            options:
                default: 1.000
```
Upgrade to v2.9 and run `doctrine:schema:validate` to reproduce.

I was able to verify that this is caused because of the "default" value set in "options". Removing this allowed Doctrine to continue to map this column as DECIMAL.

**Comment by morozov** [comment]
@mab05k please try to reproduce this issue using only DBAL. E.g.:
```php
$table = new Table('t');
$table->addColumn('column_name', 'decimal', [
    'nullable' => false,
    'precision' => 6,
    'scale' => 3,
    'default' => 1.0,
]);

var_dump($conn->getDatabasePlatform()->getCreateTableSQL($table));
```
On MySQL, the example above generatest the following both on `v2.8.0` and `v2.9.0`:
```
array(1) {
  [0] =>
  string(130) "CREATE TABLE t (column_name NUMERIC(6, 3) DEFAULT '1' NOT NULL) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB"
}
```

**Comment by alvarga** [comment]
@mab05k I had the same issue and have resolve it by changing mapping from:
<pre>
<code>
doneQty:
      column: done_qty
      type: decimal
      precision: 10
      scale: 3
      options:
        default: 0.000
</code>
</pre>

to:

<pre>
<code>
doneQty:
      column: done_qty
      type: decimal
      precision: 10
      scale: 3
      options:
        default: '0.000'
</code>
</pre>

**Comment by mab05k** [comment]
@alvarga This fix worked, thanks! 

Also, @morozov , it appears this issue has been resolved in v2.9.1.

Will close this ticket. Thanks all for your help!
